api/topology/src/main/java/quarks/topology/TStream.java

```diff
@@ -439,4 +440,11 @@ else if (WARNING.equals(lr.getLevel()))
      * @return set of tags
      */
     Set<String> getTags(); 
+    
+    <J, U, K> TStream<J> join(Function<T, K> keyer, TWindow<U, K> window, BiFunction<T, List<U>, J> joiner);
+    
+    <J, U, K> TStream<J> joinLast(Function<T, K> keyer, TStream<U> lastStream, Function<U, K> lastStreamKeyer, BiFunction<T, U, J> joiner);
+
+    <J, U> TStream<J> joinLast(TStream<U> lastStream, BiFunction<T, U, J> joiner);
```

Comment:
Why have this method? We don't provide unpartitioned variants of other methods do we?